src/EventStore.Core/Index/HashListMemTable.cs

```diff
@@ -55,14 +55,17 @@ public class HashListMemTable : IMemTable, ISearchTable {
 			SortedList<Entry, byte> list;
 			if (!_hash.TryGetValue(stream, out list)) {
 				list = new SortedList<Entry, byte>(MemTableComparer);
+				if (!Monitor.TryEnter(list, 10000))
```

Comment:
Since you're entering monitor no matter what, why not just put Monitor.TryEnter at line 56, before if, to keep the code simpler?

Also, if _hash.AddOrUpdate throws exception, you'll never exit the monitor. The if from line 56 should probably be inside try .. finally too

Reply:
@slav thanks very much for your suggestions. Putting `Monitor.TryEnter` at line 56 would not work since `list` would still be `null` at this point. However, the try-finally issue is valid, I'll push the changes shortly. thanks!